lib/config_skeleton.rb

```diff
@@ -446,6 +451,21 @@ def sleep_duration
     60
   end
 
+  # How long to ignore signals/notifications after a config regeneration
+  #
+  # Hammering a downstream service with reload requests is often a bad idea.
+  # This method exists to allow subclasses to define a 'cooldown' duration.
+  # After each config regeneration, the config generator will sleep for this
+  # duration, regardless of any CONT signals or inotify events. Those events
```

Comment:
I can't see how the events are queued up in the config_skeleton.rb class? Or is this queueing handled elsewhere?

Reply:
Also maybe make it clear that the cooldown is inside the sleep period e.g. "if your sleep duration is 60s and your cooldown is 5s then we start processing events at 5s after reload and will do so for 55s before the next loop"

Reply:
> I can't see how the events are queued up in the config_skeleton.rb class?

`IO.select()` monitors a set of IO objects until they have data available. During the cooldown period, we only watch `@terminate_r`.

`notifier.to_io`, and `@trigger_regen_r` are not being watched. But, they still exist as pipes which can fill up with data.

Then, when cooldown is done, we call IO.select on all the pipes, and it will return immediately if there is there is data ready to be read.